packages/medusa/src/services/order-edit-item-change.ts

```diff
@@ -0,0 +1,122 @@
+import { OrderItemChange } from "../models"
+import { buildQuery } from "../utils"
+import { MedusaError } from "medusa-core-utils"
+
+type InjectedDependencies = {
+  manager: EntityManager
+  orderItemChangeRepository: typeof OrderItemChangeRepository
+  eventBusService: EventBusService
+  lineItemService: LineItemService
+}
+
+export default class OrderEditItemChangeService extends TransactionBaseService {
+  static readonly Events = {
+    DELETED: "order-edit-item-change.DELETED",
+  }
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+
+  protected readonly orderItemChangeRepository_: typeof OrderItemChangeRepository
+  protected readonly eventBus_: EventBusService
+  protected readonly lineItemService_: LineItemService
+
+  constructor({
+    manager,
+    orderItemChangeRepository,
+    eventBusService,
+    lineItemService,
+  }: InjectedDependencies) {
+    // @ts-ignore
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.orderItemChangeRepository_ = orderItemChangeRepository
+    this.eventBus_ = eventBusService
+    this.lineItemService_ = lineItemService
+  }
+
+  protected async retrieve_(
+    selector: Selector<OrderItemChange>,
+    config: FindConfig<OrderItemChange> = {}
+  ): Promise<OrderItemChange> {
+    const manager = this.transactionManager_ ?? this.manager_
+    const orderItemChangeRepo = manager.getCustomRepository(
+      this.orderItemChangeRepository_
+    )
+
+    const query = buildQuery(selector, config)
+    const itemChange = await orderItemChangeRepo.findOne(query)
+
+    if (!itemChange) {
+      const selectorConstraints = Object.entries(selector)
+        .map(([key, value]) => `${key}: ${value}`)
+        .join(", ")
+
+      throw new MedusaError(
+        MedusaError.Types.NOT_FOUND,
+        `Order edit item change with ${selectorConstraints} was not found`
+      )
+    }
+
+    return itemChange
+  }
+
+  async retrieve(
+    itemChangeId: string,
+    config: FindConfig<OrderItemChange> = {}
+  ): Promise<OrderItemChange> {
+    return await this.retrieve_({ id: itemChangeId }, config)
+  }
+
+  async retrieveItemChangeByOrderEdit(
+    itemChangeId: string,
+    orderEditId: string,
+    config: FindConfig<OrderItemChange> = {}
+  ): Promise<OrderItemChange> {
+    return await this.retrieve_(
+      { id: itemChangeId, order_edit_id: orderEditId },
+      config
+    )
+  }
+
+  async delete(itemChangeIds: string | string[]): Promise<void> {
+    itemChangeIds = Array.isArray(itemChangeIds)
+      ? itemChangeIds
+      : [itemChangeIds]
+
+    return await this.atomicPhase_(async (manager) => {
+      const orderItemChangeRepo = manager.getCustomRepository(
+        this.orderItemChangeRepository_
+      )
+
+      const changes = await orderItemChangeRepo.find({
+        where: {
+          id: In(itemChangeIds as string[]),
+        },
+      })
+
+      const lineItemIdsToRemove = changes
```

Comment:
question: Should we add a migration that add cascade remove over the line item on the order item change? not the original one, just the one link to the change.

I've tried to set the relation as `cascade" ["remove"]` + call the remove method on the repository on the entities but it does not remove them.

Reply:
have no strong opinions either sounds good to me

One downside of using the cascade though could be that if we in the future want to make it possible to "replace" the `LineItemService` with a custom service (maybe one that has a different datastore), then we won't be able to benefit from the cascades. So it might be more future proof to leave it as is.

Reply:
Good point 💪 just seen that argument a day ago on the saga orchestration documentation i was reading 🤣